airflow/providers/http/operators/http.py

```diff
@@ -84,6 +88,7 @@ def __init__(
         extra_options: Optional[Dict[str, Any]] = None,
         http_conn_id: str = 'http_default',
         log_response: bool = False,
+        auth_type: Type(AuthBase) = HTTPBasicAuth,
```

Comment:
```suggestion
        auth_type: Type[AuthBase] = HTTPBasicAuth,
```

I’m surprised Mypy does not complain about this. Is this even legal?

Reply:
Actually, mypy does not complain about this interestingly enough.  I was surprised.  However, I did not re-run the provider tests on this annotation update and they do indeed fail.